core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2479,4 +2479,14 @@ package object config {
       .version("3.4.0")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val STAGE_MAX_ATTEMPTS =
+    ConfigBuilder("spark.stage.maxAttempts")
+      .doc("The max attempts for a stage, the spark job will be aborted if any of its stages is " +
+        "resubmitted multiple times beyond the limitation. The value should be no less " +
+        "than `spark.stage.maxConsecutiveAttempts` which defines the max attempts for " +
+        "fetch failures.")
+      .version("3.5.0")
+      .intConf
+      .createWithDefault(16)
```

Comment:
Since this is a behavior change, let us make this an optional paraeter - and preserve current behavior when not configured.
We can change this to a default value in a future release.

Given cascading stage retries, particularly due to `INDETERMINATE` stage, this minimizes application failures.

Reply:
Sounds good. Modified the default value to int max. Thanks.